pkg/storageincentives/agent.go

```diff
@@ -323,12 +333,24 @@ func (a *Agent) play(ctx context.Context) (uint8, []byte, error) {
 
 	storageRadius := a.radius.StorageRadius()
 
-	isPlaying, err := a.contract.IsPlaying(ctx, storageRadius)
-	if !isPlaying || err != nil {
+	// true if frozen
+	isFrozen, err := a.redistributionStatuser.IsOverlayFrozen(ctx)
+	if err != nil {
+		a.logger.Info("error checking if stake is frozen", "err", err)
+	}
+	if err == nil {
+		if isFrozen {
+			a.nodeState.SetLastFrozenRound(round)
+		}
+		a.nodeState.IsFrozen(isFrozen)
+	}
+
+	_, err = a.contract.IsPlaying(ctx, storageRadius)
```

Comment:
why is `isPlaying` boolean removed here ??
the node cannot play if isPlaying false ?

Reply:
https://github.com/ethersphere/bee/pull/3683#discussion_r1085827359

You requested this change. :)

Reply:
"we should only increment the metric on error, NOT when isPlaying is false, I know you didnt write this but can you fix it in your PR"

I don't understand what you mean. 
The current code does not even check if we are playing, I did not request this. What I requested was the metric was being incremented, that's it :)